server/transaction/manager.go

```diff
@@ -159,13 +158,15 @@ func newTxSessionWithMetrics(kv kv.KeyValueStore) (*TxSessionWithMetrics, error)
 }
 
 func (m *TxSessionWithMetrics) measure(ctx context.Context, name string, f func(ctx context.Context) error) {
-	var finishTracing func()
+	// No counters here, just tracing, will be removed in the future
 	tags := metrics.GetFdbTags(ctx, name)
 	spanMeta := metrics.NewSpanMeta(metrics.TxManagerTracingServiceName, name, "tx_manager", tags)
-	ctx, finishTracing = spanMeta.StartTracing(ctx, true)
-	defer finishTracing()
-	// TODO: better error handling (error in trace, etc)
-	ulog.E(f(ctx))
+	ctx = spanMeta.StartTracing(ctx, true)
+	err := f(ctx)
+	if err != nil {
+		spanMeta.FinishWithError(ctx, err)
+	}
+	_ = spanMeta.FinishTracing(ctx)
```

Comment:
is it ok to call this after FinishWithError?

Reply:
Good catch, the function should return after this.